Utilities/UtilData.cs

```diff
@@ -1,156 +1,201 @@
-        }
-
-        public static ushort ToUShort(byte[] Input)
-        {
-            if (Input.Length != 2) { throw new FormatException("Given byte[] does not convert to ushort."); }
-            return BitConverter.ToUInt16(EnsureBigEndian(Input), 0);
-        }
-
-        /// <summary> Returns a string from the byte representation of the string in unicode </summary>
-        /// <param name="Input"> Byte represensation of the unicode string </param>
-        /// <returns> String representation of the bytes </returns>
-        public static string ToString(byte[] Input)
-        {
-            if (Input == null || Input.Length == 0 || Input.Length % 2 == 1) { throw new FormatException("Given byte[] does not convert to string."); }
-            StringBuilder Output = new StringBuilder(Input.Length / 2);
-            for(int i = 0; i < Input.Length; i += 2)
-            {
-                Output.Append((char)(Input[i] << 8 | Input[i + 1]));
-            }
-            return Output.ToString();
-        }
-
-        /// <summary> Tries to convert a byte input (unicode) to a string </summary>
-        /// <param name="Input"> Bytes to convert to string </param>
-        /// <param name="Output"> Output of the conversion. (Null if failed) </param>
-        /// <returns> True if string conversion succeeds </returns>
-        public static bool TryToString(byte[] Input, out string Output)
-        {
-            Output = null; 
-            try { Output = ToString(Input); }
-            catch { return false; }
-            return true;
-        }
-
-        /// <summary> Determines if the given type is numeric. </summary>
-        /// <param name="Type"> Type to determine whether or not it is a numeric </param>
-        /// <returns> Returns <c>true</c> if param is a numeric; otherwise returns <c>false</c>. </returns>
-        public static bool IsNumericType(Type Type)
-        {
-            switch (Type.GetTypeCode(Type))
-            {
-                case TypeCode.Byte:
-                case TypeCode.SByte:
-                case TypeCode.UInt16:
-                case TypeCode.UInt32:
-                case TypeCode.UInt64:
-                case TypeCode.Int16:
-                case TypeCode.Int32:
-                case TypeCode.Int64:
-                case TypeCode.Decimal:
-                case TypeCode.Double:
-                case TypeCode.Single:
-                    return true;
-                default:
-                    return false;
-            }
-        }
-
-        /// <summary> Takes a 16bit number, and swaps the locations of the first and last 8b. E.g. 0x54EC would become 0xEC54. </summary>
-        /// Intended for use with 16b I2C devices that expect the byte order reversed.
-        public static ushort SwapBytes(ushort Input)
-        {
-            return (ushort)(((Input & 0b1111_1111) << 8) | ((Input >> 8) & 0b1111_1111));
-        }
-
-    }
-}
+﻿using System;

```

Comment:
Because Visual studio keeps wanting me to change line endings every time I open anything and it does this trash.

Reply:
There should be options in your Git client to normalize line endings so that this doesn't happen, but it's fine. Just makes it difficult to see what you've changed.